js/deer-record.js

```diff
@@ -69,154 +69,263 @@ export default class DeerReport {
         this.evidence = elem.getAttribute(DEER.EVIDENCE) // inherited to inputs
         this.context = elem.getAttribute(DEER.CONTEXT) // inherited to inputs
         this.type = elem.getAttribute(DEER.TYPE)
-        this.inputs = document.querySelectorAll(DEER.INPUTS.map(s=>s+"["+DEER.KEY+"]").join(","))
-
-        Array.from(this.inputs).forEach(inpt=>inpt.addEventListener('input', () => inpt.$isDirty = true))
-
+        this.inputs = elem.querySelectorAll(DEER.INPUTS.map(s => s + "[" + DEER.KEY + "]").join(","))
         changeLoader.observe(elem, {
-            attributes:true
+            attributes: true
         })
-        
         elem.oninput = event => this.$isDirty = true
         elem.onsubmit = this.processRecord.bind(this)
-        
+
         if (this.id) {
             //Do we want to expand for all types?
-            UTILS.expand({"@id":this.id})
-            //What if there are no annotations on it and the things I need to know are already in the object?
-            //Expand only returned an object like {"@id": "http://an/id"} instead of resolving it, which is what I expected. 
-            .then((function(obj){
-                Object.keys(obj).forEach((function(key){
+            UTILS.expand({ "@id": this.id })
+                .then((function(obj) {
                     try {
-                        for(let el of Array.from(this.inputs)) {
-                            if(el.getAttribute(DEER.KEY)===key){
-                                let assertedValue = UTILS.getValue(obj[key])
-                                if(Array.isArray(assertedValue)) {
-                                    for (const v of assertedValue) {
-                                        if(!el.value && (["string","number"].indexOf(typeof v)!==-1)){
-                                            el.value = v
+                        let inputElems = Array.from(this.inputs)
+                        let flatKeys = new Set(inputElems.map(input => input.getAttribute(DEER.KEY)))
+                        for (let el of inputElems) {
+                            let deerKeyValue = (el.hasAttribute(DEER.KEY)) ? el.getAttribute(DEER.KEY) : ""
+                            let mapsToAnno = false
+                            if (deerKeyValue) {
+                                //Then this is a DEER form input.
+                                el.addEventListener('input', (e) => e.target.$isDirty = true)
+                                let assertedValue = ""
+                                if (flatKeys[deerKeyValue] !== i) {
+                                    UTILS.warning("Duplicate input " + DEER.KEY + " attribute value '" + deerKeyValue + "' detected in form.  This input will be ignored upon form submission and only the first instance will be respected.  See duplicate below.", el)
+                                    el.setAttribute(DEER.KEYDUPLICATE, "true")
```

Comment:
This is a fringe I did not really think about.  In my mind, neither of these would have a DEER.KEY.  I would have made it so that a change to either of them would write to a hidden `<input>` that has DEER.KEY="priority" so that the annotation could not be confused or duplicated.  DEER can only know to make an annotation if DEER.KEY exists, and if DEER.KEY is repeated how is it supposed to know whether to duplicate or not?  What could the dev do to make it understand that it isn't supposed to make this annotation twice?

I think this is one of those "How DEER should be used" moments that we will keep coming across.  I am open to ideas for how to handle more complex forms like this.